### PR TITLE
Fixed Organization Post Card

### LINKED ISSUE #1713: Video Post Creation - DOMException: The play() request was interrupted by a call to pause()
**Describe the bug**
When creating a post with a video upload and subsequently viewing the post, the following error occurs in the console: "Uncaught (in promise) DOMException: The play() request was interrupted by a call to pause()."

**To Reproduce**

1. Create a post with a video upload.
2. View the created post.
3. Open the browser console.

**Expected behavior**
The video should play without any interruptions or errors when viewing the post.

**Screenshots**
image

**Additional details**

**Potential internship candidates**
Please read this if you are planning to apply for a Palisadoes Foundation internship https://github.com/PalisadoesFoundation/talawa/issues/359

### PR DESCRIPTION
**Pull request by Shreyas2309**
**What kind of change does this PR introduce?**

Bug Fixed Feature added

**Issue Number:**

Fixes #1713 

**Did you add tests for your changes?**

N/A

**Snapshots/Videos:**

https://vimeo.com/923604484?share=copy

**If relevant, did you update the documentation?**

N/A

**Summary**

1. What was the problem ?

   The DOMException: The play() request was interrupted by a call to pause() error typically occurs in web development when 
   there's an attempt to play media (like audio or video) that gets interrupted by a call to pause the media playback. This can 
   happen due to various reasons, such as asynchronous events or user interactions triggering both play and pause commands 
   simultaneously.

2. Motivation for solving the problem .

    The motivation behind solving this issue through a pull request is to enhance the robustness and reliability of web 
    applications, particularly those involving media playback. By addressing this error, developers aim to provide a smoother and 
    more seamless user experience by preventing unexpected interruptions in media playback. 

3. After solving the issue....
   
   By resolving this issue, developers can ensure that media playback behaves predictably and as intended, reducing the 
   likelihood of errors mentioned and improving the overall quality of the user experience. #1713 

**Does this PR introduce a breaking change?**

N/A

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by dhanagopu** [comment]
@anwersayeed @beingnoble03 Can you review this PR?

**Comment by dhanagopu** [comment]
@Shreyas2309 Please fix the conflicting files.

**Comment by Shreyas2309** [comment]
I think I have Resolved the conflicts. 
Please review the PR.

File: src/components/OrgPostCard/OrgPostCard.tsx [hunk 1]
```diff
@@ -89,15 +89,33 @@ export default function orgPostCard(
...
   };
 
   const handleVideoPause = (): void => {
-    setPlaying(false);
-    videoRef.current?.pause();
+    // setPlaying(false);
+    // videoRef.current?.pause();
+    // setModalVisible(false);
+    if (playing) {
+      console.log('IIIII');
```
Comment (beingnoble03) [inline src/components/OrgPostCard/OrgPostCard.tsx:111]: remove debug statement?

Comment (beingnoble03) [inline src/components/OrgPostCard/OrgPostCard.tsx:111]: remove debug statement?

File: src/components/OrgPostCard/OrgPostCard.tsx [hunk 2]
```diff
@@ -89,15 +89,33 @@ export default function orgPostCard(
...
+        });
+    }
   };
 
   const handleVideoPause = (): void => {
-    setPlaying(false);
-    videoRef.current?.pause();
+    // setPlaying(false);
+    // videoRef.current?.pause();
+    // setModalVisible(false);
```
Comment (beingnoble03) [inline src/components/OrgPostCard/OrgPostCard.tsx:109]: remove comments.

File: src/components/OrgPostCard/OrgPostCard.tsx [hunk 3]
```diff
@@ -89,15 +89,33 @@ export default function orgPostCard(
...
 
   const handleVideoPlay = (): void => {
-    setPlaying(true);
-    videoRef.current?.play();
+    const playPromise = videoRef.current?.play();
+
+    if (playPromise !== undefined) {
+      playPromise
+        .then(() => {
+          // setModalVisible(true);
```
Comment (beingnoble03) [inline src/components/OrgPostCard/OrgPostCard.tsx:97]: remove comment

File: src/components/OrgPostCard/OrgPostCard.tsx [hunk 4]
```diff
@@ -397,7 +413,9 @@ export default function orgPostCard(
               </button>
               <button
                 className={styles.closeButton}
-                onClick={(): void => setModalVisible(false)}
+                onClick={(): void => {
+                  setModalVisible(false);
+                }}
```
Comment (beingnoble03) [inline src/components/OrgPostCard/OrgPostCard.tsx:418]: seems unnecessary change.

File: package-lock.json
Comment (beingnoble03) [inline package-lock.json:1]: i don't think we need package-lock and package.json changes here?

File: package.json
Comment (beingnoble03) [inline package.json:1]: i don't think we need package-lock and package.json changes here?

**Review by beingnoble03 — CHANGES_REQUESTED** [review]
there are some lint errors, can you please fix those?

**Comment by akhilender-bongirwar** [comment]
checknotrunning


@beingnoble03, I think the check `PR Workflow / Performs linting, formatting, type-checking (pull_request) ` is not working fine as there are no files with changes more than 600 lines in this PR.

**Comment by Shreyas2309** [comment]
I have fixed the linting errors . Can you please review it.

File: src/components/OrgPostCard/OrgPostCard.tsx (hunk 1 above)
Comment (Shreyas2309) [reply src/components/OrgPostCard/OrgPostCard.tsx]: removed it

File: src/components/OrgPostCard/OrgPostCard.tsx (hunk 2 above)
Comment (Shreyas2309) [reply src/components/OrgPostCard/OrgPostCard.tsx]: removed it

File: src/components/OrgPostCard/OrgPostCard.tsx (hunk 3 above)
Comment (Shreyas2309) [reply src/components/OrgPostCard/OrgPostCard.tsx]: removed it

File: src/components/OrgPostCard/OrgPostCard.tsx (hunk 4 above)
Comment (Shreyas2309) [reply src/components/OrgPostCard/OrgPostCard.tsx]: fixed it

File: package-lock.json
Comment (Shreyas2309) [reply package-lock.json:1]: removed the unnecessary changes

File: package.json
Comment (Shreyas2309) [reply package.json:1]: removed the unnecessary changes

**Comment by palisadoes** [comment]
We are going to pause merging PRs into the `develop` branch to facilitate fixing the roles of Members, Admins and Super Admins.

They need to assigned on a per organization basis versus per API instance. This is the tracking issue.

- https://github.com/PalisadoesFoundation/talawa-api/issues/1965

Reviews of PRs for all branches will continue

**Comment by palisadoes** [comment]
This is an update on the PR merging freeze:

1. In the next few hours we will be merging the `develop-userTypeFix` branch into the `develop` branch. 
1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. This merge will cause some conflicts in your PR.

We decided to do this at the beginning of the weekend to give us all time to adjust PR code and create bug fixes that may arise.

Update your code at or after midnight GMT on the morning of March 23, 2024. (5:30am IST). 

If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Comment by palisadoes** [comment]
## The PR merging freeze is lifted.

1. We are working on bug fixes that may arise. These should be few and are being tracked here:
    1. https://github.com/orgs/PalisadoesFoundation/projects/24
1. Please update your PRs and local repos. Fix any new merge conflicts that may have occurred.
    
Background:

1. The `develop-userTypeFix` branch was created to fix a long standing design flaw where Admins were Admins of all organizations, not specific ones.
1. The `userType` field has been removed from the User collection and it has been replaced by an `appUserProfileId` field.
    1. This field is `null` if the user isn’t registered to use the apps. This will help people to add users manually during the event checkin process, or if an Admin wants to manually add someone in the Admin dashboard.
    1. When not `null` the `AppUserProfileID` will reference a `AppUserProfile` collection with App related information such as the organizations for which a user may be an Admin.
    1. The updated schema can be found here https://github.com/PalisadoesFoundation/talawa-api/blob/develop-userTypeFix/schema.graphql
    1. This is the parent issue that we have been using to track progress:
        1. https://github.com/PalisadoesFoundation/talawa-api/issues/1965
1. If your PRs have already been approved, request a re-review after fixing the conflicts and refactoring to the new `AppUserProfileID` methodology.

**Comment by Cioppolo14** [comment]
@Shreyas2309 Please fix the failing tests and conflicting files.

**Comment by palisadoes** [comment]
Closing as abandoned

**Pull request closed without merging**